src/widgets/chart.rs

```diff
@@ -385,28 +408,77 @@ impl<'a> Chart<'a> {
         if labels_len < 2 {
             return;
         }
-        let width_between_ticks = graph_area.width / (labels_len - 1);
+
+        let width_between_ticks = graph_area.width / labels_len;
+
         for (i, label) in labels.iter().enumerate() {
-            let label_width = label.width() as u16;
-            let label_width = if i == 0 {
-                // the first label is put between the left border of the chart and the y axis.
-                graph_area
-                    .left()
-                    .saturating_sub(chart_area.left())
-                    .min(label_width)
+            if i == 0 {
```

Comment:
we return early if labels_len < 2 therefore wouldn't it be more clear to unroll this loop:
```rust
// first label

// labels[1..labels.len() - 1]

// last label
```

Reply:
You're right, the loop becomes much clearer once we exclude the first and last label cases from it.